sycl/include/sycl/usm.hpp

```diff
@@ -156,15 +290,15 @@ template <typename T>
 T *malloc_device(
     size_t Count, const device &Dev, const context &Ctxt,
     const property_list &PropList = {},
-    const detail::code_location CodeLoc = detail::code_location::current()) {
+    const detail::code_location &CodeLoc = detail::code_location::current()) {
```

Comment:
I think template versions must use `#ifndef DISABLE_SYCL_INSTRUMENTATION_METADATA` too. Right?

Reply:
I think so too, PR was just growing too large. Updated it.

Reply:
I'm okay to separate this change form this PR if it helps.